homeassistant/components/supla/__init__.py

```diff
@@ -98,7 +98,7 @@ async def discover_devices(hass, hass_config):
 
     Currently it is only run at startup.
     """
-    component_configs: dict[Platform, list[dict]] = {}
+    component_configs: dict[Platform, dict] = {}
 
     for server_name, server in hass.data[DOMAIN][SUPLA_SERVERS].items():
```

Comment:
We're collecting the channels for potentially multiple servers so we need to create the key from both the server name and the channel id, to put in `component_configs`.